### ISSUE TITLE
URL to mediaconvert getJobTemplate url becomes double escaped 

### ISSUE THREAD

**Issue by mansatgigset** [awslabs/aws-sdk-swift#1043]
### Describe the bug

let template = try await mediaConvert.getJobTemplate(input: .init("Android TV Template"))

info CRTClientEngine : [Logging] Connection was acquired to: Optional("https://gnwesle3b.mediaconvert.eu-north-1.amazonaws.com/2017-08-29/jobTemplates/Android%2520TV%2520Template?")

Causing the request to fail

### Expected Behavior

For the called to work

### Current Behavior

The call fails due invalid escaping.

### Reproduction Steps

let template = try await mediaConvert.getJobTemplate(input: .init("Android TV Template"))

info CRTClientEngine : [Logging] Connection was acquired to: Optional("https://gnwesle3b.mediaconvert.eu-north-1.amazonaws.com/2017-08-29/jobTemplates/Android%2520TV%2520Template?")

Causing the request to fail

### Possible Solution

Update this

```swift
extension GetJobTemplateInput: ClientRuntime.URLPathProvider {
    public var urlPath: Swift.String? {
        guard let name = name else {
            return nil
        }
        return "/2017-08-29/jobTemplates/\(name).urlPercentEncoding())"
    }
}
```

to 

```swift
extension GetJobTemplateInput: ClientRuntime.URLPathProvider {
    public var urlPath: Swift.String? {
        guard let name = name else {
            return nil
        }
        return "/2017-08-29/jobTemplates/\(name)"
    }
}
```

### AWS SWIFT SDK version used

0.18.0

### Compiler and Version used

Xcode 14.3.1 (14E300c)

### Operating System and version

Mac Ventura 13.4 (22F66)

**Comment by jbelkins** [comment]
Thanks for your report.  I'll look into this and see if I can come up with a fix.  I'll inform you here when I have something ready for you to try.

**Comment by jbelkins** [comment]
@mansatgigset To be clear, was this the error you received?
```
BadRequestException(properties: AWSMediaConvert.BadRequestException.Properties(message: Optional("\'Android%20TV%20Template\' is not a valid name, the maximum length for a name is 80, and the characters $&,:;?<>`\"#%{}/|\\^~ are not allowed.")), httpResponse: <omitted>)
```

**Comment by mansatgigset** [comment]
@jbelkins Yes that is correct, I should of course have added that one in the original issue.

**Comment by mansatgigset** [comment]
@jbelkins Thanks for the quick response and solution.

**Comment by jbelkins** [comment]
Keeping this ticket open until the fix releases

**Comment by jbelkins** [comment]
Shipped in AWS SDK for Swift 0.20.0
